### PR TITLE
chore: add a twitter compose ruleset

### PR DESCRIPTION
**Pull request by trOnk12**
## Description
- To make sure we are not doing any jetpack compose pitfalls, there is a twitter compose ruleset that we can use.

https://twitter.github.io/compose-rules/rules/

### Why?
- Add the twitter compose ruleset to be ran by ktlint to our custom rules set.

## Links to related issues

- Fixes [LD-XXX](https://loudius.atlassian.net/jira/software/projects/LD/boards/3)
- Not covered by the PR: [LD-XXX](https://loudius.atlassian.net/jira/software/projects/LD/boards/3)
- Possible improvements: [LD-XXX](https://loudius.atlassian.net/jira/software/projects/LD/boards/3)

## Demo

| Before the PR | After the PR |
|---------------|--------------|
|               |              |

## Documentation
- Link to documentation: **Add link here**
- Link to figma: **Add link here**
- **Add here other useful documentation links that are useful for PR reviewer**

## Checklist

- [ ] New functionality is covered by Analytics events
- [ ] Analytics events are documented
- [ ] Analytics dashboard is updated
- [ ] Functionality covered by unit tests
- [ ] I have manually tested if the code works

### CODE REVIEW

File: custom-ktlint-rules/build.gradle [hunk 1]
```diff
@@ -4,16 +4,17 @@ plugins {
 }
 
 java {
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
```
Comment (trOnk12) [inline custom-ktlint-rules/build.gradle]: it was required by twitter compose rules

**Comment by trOnk12** [comment]
not working as for now ;s

File: .github/workflows/run-code-quality-check.yml [hunk 2]
```diff
@@ -56,7 +56,7 @@ jobs:
           # https://megalinter.github.io/configuration/
           VALIDATE_ALL_CODEBASE: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }} # Validates all source when push on main, else just the git diff with main. Override with true if you always want to lint all sources
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          KOTLIN_KTLINT_ARGUMENTS: "--ruleset=custom-ktlint-rules/build/libs/custom-ktlint-rules.jar"
+          KOTLIN_KTLINT_ARGUMENTS: "--ruleset=com.twitter.compose.rules:ktlint:0.0.26"
```
Comment (jacek-marchwicki) [inline .github/workflows/run-code-quality-check.yml]: **bug**: You've removed our custom-ktlint-rules, you should keep using our custom rules.

**Review by jacek-marchwicki — CHANGES_REQUESTED** [review]

File: .github/workflows/run-code-quality-check.yml (hunk 2 above)
Comment (trOnk12) [reply .github/workflows/run-code-quality-check.yml]: this is a draft as for now for testing, ill let you know once it is ready for review.

File: custom-ktlint-rules/build.gradle (hunk 1 above)
Comment (trOnk12) [reply custom-ktlint-rules/build.gradle]: not anymore, reverted it back to what it was

**Comment by trOnk12** [comment]
closing due to being unable to resolve the issues with adding the twitter compose rules

**Pull request closed without merging**